### PR TITLE
Fix race condition preventing the app opened from a PN to go to the correct channel

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Refactored code around the entry js file to fix a race condition preventing a push notification opening in the right channel if the app was closed.

This PR also removes the `LoadTeam` screen as it is not needed anymore and make sure that EMM is also handled and not called twice as it was before.

#### Ticket Link
https://mattermost.atlassian.net/browse/ICU-577

#### Device Information
This PR was tested on: 
* Android 7.1.1 Nexus 5 (Emulator)
* Android 6.0.1 Samsung J5 (Device)
* iOS 11.2 iPhone X (Simulator)
* iOS 11.2.5 iPhone 6s (Device)

### CODE REVIEW

File: app/actions/views/login.js [hunk 1]
```diff
@@ -39,9 +39,6 @@ export function handleSuccessfulLogin() {
             }
         }, getState);
 
-        Client.setToken(token);
```
Comment (hmhealey) [inline app/actions/views/login.js]: Are we not using the regular Client at all for anything now, or is this set somewhere else?

File: app/components/root/index.js [hunk 2]
```diff
@@ -15,7 +15,6 @@ import Root from './root';
 
 function mapStateToProps(state) {
     const locale = getCurrentLocale(state);
-    Client.setLocale(locale);
```
Comment (hmhealey) [inline app/components/root/index.js]: Same thing here

File: app/mattermost.js [hunk 3]
```diff
@@ -6,14 +6,7 @@ import Orientation from 'react-native-orientation';
...
 import {IntlProvider} from 'react-intl';
-import {
-    Alert,
-    AppState,
-    InteractionManager,
-    Keyboard,
-    NativeModules,
-    Platform
-} from 'react-native';
+import {Alert, AppState, InteractionManager, Keyboard, NativeModules, Platform} from 'react-native';
```
Comment (hmhealey) [inline app/mattermost.js]: You don't need to change these back now, but can we try to have these spread across multiple lines when we're importing this many things? It makes it easier to read and see changes

File: app/mattermost.js [hunk 4]
```diff
@@ -229,20 +200,20 @@ export default class Mattermost {
...
         if (serverVersion) {
-            if (semver.valid(version) && semver.lt(version, Config.MinServerVersion)) {
+            if (semver.valid(version) && semver.lt(version, LocalConfig.MinServerVersion)) {
                 Alert.alert(
                     intl.formatMessage({id: 'mobile.server_upgrade.title', defaultMessage: 'Server upgrade required'}),
                     intl.formatMessage({id: 'mobile.server_upgrade.description', defaultMessage: '\nA server upgrade is required to use the Mattermost app. Please ask your System Administrator for details.\n'}),
                     [{
                         text: intl.formatMessage({id: 'mobile.server_upgrade.button', defaultMessage: 'OK'}),
-                        onPress: this.handleVersionUpgrade
+                        onPress: this.handleAppVersionUpgraded
```
Comment (hmhealey) [inline app/mattermost.js]: Is this an app version upgrade or a server version upgrade? The help text sounds like a server one

File: app/actions/views/login.js (hunk 1 above)
Comment (enahum) [reply app/actions/views/login.js]: We don't use the regular Client at all in the RN app

**Comment by enahum** [comment]
@hmhealey changes made

**Pull request merged**